src/main/java/uk/gov/justice/digital/delius/transformers/AppointmentPatchRequestTransformer.java

```diff
@@ -0,0 +1,100 @@
+package uk.gov.justice.digital.delius.transformers;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.fge.jsonpatch.JsonPatch;
+import com.github.fge.jsonpatch.JsonPatchOperation;
+import com.github.fge.jsonpatch.ReplaceOperation;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Component;
+import uk.gov.justice.digital.delius.config.DeliusIntegrationContextConfig.IntegrationContext;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.StreamSupport;
+
+import static com.fasterxml.jackson.databind.node.TextNode.valueOf;
+import static com.github.fge.jackson.jsonpointer.JsonPointer.of;
+import static java.lang.String.format;
+import static java.util.Optional.ofNullable;
+import static java.util.stream.Collectors.toList;
+
+@Component
+@AllArgsConstructor
+public class AppointmentPatchRequestTransformer {
+
+    private static final String JSON_PATCH_PATH_FIELD = "path";
+    private static final String JSON_PATCH_VALUE_FIELD = "value";
+
+    private static final String SOURCE_ATTENDED_FIELD_PATH = "/attended";
+    private static final String SOURCE_NOTIFY_PP_FIELD_PATH = "/notifyPPOfAttendanceBehaviour";
+    private static final String TARGET_OUTCOME_FIELD_NAME = "outcome";
+
+    private final ObjectMapper objectMapper;
+
+    public JsonPatch mapAttendanceFieldsToOutcomeOf(final JsonPatch jsonPatch, final IntegrationContext context) {
+
+        var nodes = objectMapper.convertValue(jsonPatch, JsonNode.class);
```

Comment:
see you're converting to json here & using the jackson api to mutate the patch document. Is this required? can't you just copy & mutate the JsonPatch structure in place? I feel a lot of the code here is spent on navigating the serialised json patch structure for which we have this JsonPatch abstraction.

Reply:
ergh I just had a look at the API for JsonPatch and I'm not sure this is possible. What a shame.

Reply:
Found very little support for modifying a JsonPatch in the way we wanted. Will talk to you about what we can do instead.

Reply:
Complexity now removed as source is a DTO